metricbeat/module/docker/network_summary/helper.go

```diff
@@ -55,19 +54,19 @@ func fetchContainerNetStats(client *client.Client, timeout time.Duration, contai
 
 	inspect, err := client.ContainerInspect(ctx, container)
 	if err != nil {
-		return nil, errors.Wrapf(err, "error fetching stats for container %s", container)
+		return nil, fmt.Errorf("error fetching stats for container %s: %w", container, err)
 	}
 	rootPID := inspect.ContainerJSONBase.State.Pid
 
 	proc, err := sysinfo.Process(rootPID)
 	procNet, ok := proc.(sysinfotypes.NetworkCounters)
 	if !ok {
-		return nil, errors.Wrapf(err, "cannot fetch network counters for PID %d", rootPID)
+		return nil, fmt.Errorf("cannot fetch network counters for PID %d: %w", rootPID, err)
```

Comment:
do we have a potential err = nil here?

Reply:
@cmacknz ^^

Reply:
It looks like on !linux, the return from `sysinfo.Process` does not implement `types.NetworkCounters`, so yes, we can have a nil error here.

Reply:
fixed